src/server/game/Spells/Spell.cpp

```diff
@@ -2262,6 +2262,16 @@ void Spell::AddDestTarget(SpellDestination const& dest, uint32 effIndex)
     m_destTargets[effIndex] = dest;
 }
 
+bool Spell::TargetInfo::MissedTarget(Spell *spell)
+{
+    Unit* unit = spell->m_caster->GetGUID() == TargetGUID ? spell->m_caster->ToUnit() : ObjectAccessor::GetUnit(*spell->m_caster, TargetGUID);
+    if (!unit)
+        return true;
+    if (unit->HasStealthAura() && unit->IsHostileTo(spell->m_caster))
```

Comment:
Mass dispel should still hit rogues in stealth, remove a buff, and pulse combat. You're breaking it.